core/src/main/java/com/datastrato/gravitino/authorization/AccessControlManager.java

```diff
@@ -34,27 +35,22 @@
 import java.util.Map;
 
 /**
- * AccessControlManager is used for manage users, roles, admin, grant information, this class is an
- * entrance class for tenant management. This lock policy about this is as follows: First, admin
- * operations are prevented by one lock. Then, other operations are prevented by the other lock. For
- * non-admin operations, Apache Gravitino doesn't choose metalake level lock. There are some reasons
- * mainly: First, the metalake can be renamed by users. It's hard to maintain a map with metalake as
- * the key. Second, the lock will be couped with life cycle of the metalake.
+ * AccessControlManager is used for manage users, roles, grant information, this class is an
+ * entrance class for tenant management. The operations will be protected by one lock.
  */
 public class AccessControlManager {
 
   private final UserGroupManager userGroupManager;
-  private final AdminManager adminManager;
   private final RoleManager roleManager;
   private final PermissionManager permissionManager;
-  private final Object adminOperationLock = new Object();
-  private final Object nonAdminOperationLock = new Object();
+  private final Object operationLock = new Object();
```

Comment:
I feel like we can use tree lock instead of global lock here. For example, you can use user namespace write lock for addUser, and also for group, we can have another tree lock for group.

Reply:
I changed the lock to treeLock.